File: Array/check_permutation.py
def check_permutation(s1: str, s2: str) -> bool:
    if len(s1) != len(s2):
        return False
    char_set = {}
    for i in s1:
        if i in char_set.keys():
            count = char_set[i]
            char_set[i] = count + 1
        else:
            char_set[i] = 1
    print(char_set)
    for i in s2:
        if i in char_set.keys():
            count = char_set[i]
            char_set[i] = count-1
            if char_set[i] < 0:
                print(char_set)
                print("Count less than zero")
                return False
        else:
            return False
    for i in char_set.values():
        if i != 0:
            return False
    return True


print(check_permutation("abc", "aaa"))

File: Array/check_permutation.js
const check_permutation = (s1,s2)=>{
    console.log(s1,s2);
    if(s1.length !== s2.length){
        return false;
    }
    let char_set = new Map();
    for(let i of s1){
        if(char_set.has(i)){
            let count = char_set.get(i)
            char_set.set(i,count+1)
        }
        else{

            char_set.set(i,1)
        }
    }
    console.table(char_set)
    for(let i of s2){
        if(char_set.has(i)){
            let count = char_set.get(i)
            char_set.set(i,count-1)
        }
        else{

        return false;        }
    }
    console.table(char_set)
    for(let i of char_set.values()){
        console.log(i)
        if(i!==0){
            return false;
        }
    }
    return true;
}

console.log(check_permutation("abcaa","abcaa"))